src/packages/utils/tests/request-coordinator.test.ts

```diff
@@ -0,0 +1,98 @@
+import assert from "assert";
+import { RequestCoordinator } from "../src/utils/request-coordinator";
+
+describe("request-coordinator", () => {
+  const noop = () => undefined;
+  let coordinator: RequestCoordinator;
+
+  beforeEach("instantiate RequestCoordinator", () => {
+    coordinator = new RequestCoordinator(0);
+  });
+
+  describe("stop()", () => {
+    it("should pause processing", () => {
```

Comment:
The test doesn't actually test if processing is paused, it just tests that calling `stop` sets the `paused` property to `true`.

```suggestion
    it("should set `paused` property to `true`", () => {
```

Reply:
Good point!